From 31045601a05ec64ecf94a605d12b4f81bc12bc52 Mon Sep 17 00:00:00 2001
Message-Id: <31045601a05ec64ecf94a605d12b4f81bc12bc52.1430272235.git.chang-joon.lee@intel.com>
In-Reply-To: <0282bce11918ec0dd0c4e7e3bd54e392c2b27a51.1430272235.git.chang-joon.lee@intel.com>
References: <0282bce11918ec0dd0c4e7e3bd54e392c2b27a51.1430272235.git.chang-joon.lee@intel.com>
From: Uma Shankar <uma.shankar@intel.com>
Date: Fri, 10 Apr 2015 19:25:33 +0530
Subject: [PATCH 06/10] SQUASHME! [VPG]: drm/i915: Added wait for DSI PLL lock

Added a waiting loop for DSI PLL to lock. Timeout is set for
20ms.

Squash this patch to:
	Author: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
	Date:   Thu Oct 16 18:59:37 2014 +0530
	FOR_UPSTREAM [VPG]: drm/i915: program DSI PLL from intel_dsi pixel clk
	Change-Id: Id271edd570da2f0b099ffa1b25c78fcd282bf0b0

Issue: GMINL-8141
Change-Id: Ifcabd00c7a234073dabf557c713e73c159e7e822
Signed-off-by: Uma Shankar <uma.shankar@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi_pll.c |   12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dsi_pll.c b/drivers/gpu/drm/i915/intel_dsi_pll.c
index 23733ad..064279a 100644
--- a/drivers/gpu/drm/i915/intel_dsi_pll.c
+++ b/drivers/gpu/drm/i915/intel_dsi_pll.c
@@ -326,13 +326,15 @@ void vlv_enable_dsi_pll(struct intel_encoder *encoder)
 	tmp |= DSI_PLL_VCO_EN;
 	vlv_cck_write(dev_priv, CCK_REG_DSI_PLL_CONTROL, tmp);
 
+	if (wait_for(vlv_cck_read(dev_priv, CCK_REG_DSI_PLL_CONTROL) &
+							0x1, 20)) {
+		mutex_unlock(&dev_priv->dpio_lock);
+		DRM_ERROR("DSI PLL lock failed\n");
+		return;
+	}
 	mutex_unlock(&dev_priv->dpio_lock);
 
-	tmp = vlv_cck_read(dev_priv, CCK_REG_DSI_PLL_CONTROL);
-	if (tmp & 0x1)
-		DRM_DEBUG_KMS("DSI PLL Locked\n");
-	else
-		DRM_DEBUG_KMS("DSI PLL lock failed\n");
+	DRM_DEBUG_KMS("DSI PLL Locked\n");
 }
 
 void vlv_disable_dsi_pll(struct intel_encoder *encoder)
-- 
1.7.9.5

